bundles/org.openhab.binding.electroluxappliances/src/main/java/org/openhab/binding/electroluxappliances/internal/handler/ElectroluxAppliancesBridgeHandler.java

```diff
@@ -0,0 +1,191 @@
+import org.openhab.binding.electroluxappliances.internal.ElectroluxAppliancesBridgeConfiguration;
+import org.openhab.binding.electroluxappliances.internal.api.ElectroluxGroupAPI;
+import org.openhab.binding.electroluxappliances.internal.discovery.ElectroluxAppliancesDiscoveryService;
+import org.openhab.binding.electroluxappliances.internal.dto.ApplianceDTO;
+import org.openhab.binding.electroluxappliances.internal.listener.TokenUpdateListener;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * The {@link ElectroluxAppliancesBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class ElectroluxAppliancesBridgeHandler extends BaseBridgeHandler implements TokenUpdateListener {
+
+    private final Logger logger = LoggerFactory.getLogger(ElectroluxAppliancesBridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(THING_TYPE_BRIDGE);
+
+    private int refreshTimeInSeconds = 300;
+    private boolean isCommunicationError = false;
+
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private final Map<String, ApplianceDTO> electroluxAppliancesThings = new ConcurrentHashMap<>();
+
+    private @Nullable ElectroluxGroupAPI api;
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable ScheduledFuture<?> instantUpdate;
+
+    public ElectroluxAppliancesBridgeHandler(Bridge bridge, HttpClient httpClient, Gson gson) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = gson;
+    }
+
+    @Override
+    public void initialize() {
+        ElectroluxAppliancesBridgeConfiguration config = getConfigAs(ElectroluxAppliancesBridgeConfiguration.class);
+
+        ElectroluxGroupAPI electroluxGroupAPI = new ElectroluxGroupAPI(config, gson, httpClient, this);
+        refreshTimeInSeconds = config.refresh;
+
+        if (config.apiKey == null || config.accessToken == null || config.refreshToken == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Configuration of API key, access and refresh token is mandatory");
+        } else if (refreshTimeInSeconds < 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Refresh time cannot be negative!");
```

Comment:
According to the thing-types.xml
```suggestion
        } else if (refreshTimeInSeconds < 10) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                    "Refresh time cannot be less then 10!");
```